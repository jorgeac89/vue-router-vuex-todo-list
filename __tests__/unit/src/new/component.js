import { createLocalVue, RouterLinkStub, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import New from 'new';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('New component', () => {
  it('renders correctly without text', () => {
    const store = new Vuex.Store({
      modules: {
        new: {
          namespaced: true,
          state: {
            todoText: ''
          },
          actions: {
            addTodo: jest.fn()
          }
        }
      }
    });
    expect(
      shallowMount(New, {
        store,
        localVue,
        stubs: {
          RouterLink: RouterLinkStub
        }
      })
    ).toMatchSnapshot();
  });

  it('renders correctly with text', () => {
    const store = new Vuex.Store({
      modules: {
        new: {
          namespaced: true,
          state: {
            todoText: 'Example todo'
          },
          actions: {
            addTodo: jest.fn()
          }
        }
      }
    });
    expect(
      shallowMount(New, {
        store,
        localVue,
        stubs: {
          RouterLink: RouterLinkStub
        }
      })
    ).toMatchSnapshot();
  });
});
