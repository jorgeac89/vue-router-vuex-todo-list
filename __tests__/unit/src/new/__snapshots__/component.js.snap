// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New component renders correctly with text 1`] = `
<div>
  <div class="row">
    <div class="col-12 d-flex align-items-center justify-content-between">
      <h2>
        New todo
      </h2> <a class="font-weight-bold"><i class="fas fa-list-alt mr-2"></i>
        Todo list
      </a>
    </div>
  </div>
  <form>
    <div class="row mt-3">
      <div class="col-12"><textarea placeholder="Write the todo text here..." class="form-control form-control-lg mb-2"></textarea></div>
    </div>
    <div class="row mt-3">
      <div class="col-12 text-center"><button type="button" class="btn btn-primary font-weight-bold"><i class="fas fa-save mr-2"></i>
          Save
        </button></div>
    </div>
  </form>
</div>
`;

exports[`New component renders correctly without text 1`] = `
<div>
  <div class="row">
    <div class="col-12 d-flex align-items-center justify-content-between">
      <h2>
        New todo
      </h2> <a class="font-weight-bold"><i class="fas fa-list-alt mr-2"></i>
        Todo list
      </a>
    </div>
  </div>
  <form>
    <div class="row mt-3">
      <div class="col-12"><textarea placeholder="Write the todo text here..." class="form-control form-control-lg mb-2"></textarea></div>
    </div>
    <div class="row mt-3">
      <div class="col-12 text-center"><button disabled="disabled" type="button" class="btn btn-primary font-weight-bold"><i class="fas fa-save mr-2"></i>
          Save
        </button></div>
    </div>
  </form>
</div>
`;
